impeller/renderer/backend/vulkan/pipeline_vk.cc

```diff
@@ -142,9 +142,7 @@ static vk::UniqueRenderPass CreateCompatRenderPassForPipeline(
                                       LoadAction::kDontCare,       //
                                       StoreAction::kDontCare       //
     );
-  }
-
-  if (desc.HasStencilAttachmentDescriptors()) {
+  } else if (desc.HasStencilAttachmentDescriptors()) {
```

Comment:
Wait, won't this just turn off the stencil attachment and make legacy clips + StC not work anywhere?

I think the depth and stencil are separately configurable attachments even if they share a texture. Were you getting validation errors before this?

Reply:
IIRC Depth+Stencil are the same attachment in Vulkan. Calling SetStencilAttachment blows away the value of SetDepthStencilAttachment at least in our code

```
RenderPassBuilderVK& RenderPassBuilderVK::SetDepthStencilAttachment(
    PixelFormat format,
    SampleCount sample_count,
    LoadAction load_action,
    StoreAction store_action) {
  vk::AttachmentDescription desc;
  desc.format = ToVKImageFormat(format);
  desc.samples = ToVKSampleCount(sample_count);
  desc.loadOp = ToVKAttachmentLoadOp(load_action);
  desc.storeOp = ToVKAttachmentStoreOp(store_action);
  desc.stencilLoadOp = desc.loadOp;    // Not separable in Impeller.
  desc.stencilStoreOp = desc.storeOp;  // Not separable in Impeller.
  desc.initialLayout = vk::ImageLayout::eGeneral;
  desc.finalLayout = vk::ImageLayout::eGeneral;
  depth_stencil_ = desc;
  return *this;
}

RenderPassBuilderVK& RenderPassBuilderVK::SetStencilAttachment(
    PixelFormat format,
    SampleCount sample_count,
    LoadAction load_action,
    StoreAction store_action) {
  vk::AttachmentDescription desc;
  desc.format = ToVKImageFormat(format);
  desc.samples = ToVKSampleCount(sample_count);
  desc.loadOp = vk::AttachmentLoadOp::eDontCare;
  desc.storeOp = vk::AttachmentStoreOp::eDontCare;
  desc.stencilLoadOp = ToVKAttachmentLoadOp(load_action);
  desc.stencilStoreOp = ToVKAttachmentStoreOp(store_action);
  desc.initialLayout = vk::ImageLayout::eGeneral;
  desc.finalLayout = vk::ImageLayout::eGeneral;
  depth_stencil_ = desc;
  return *this;
}
```

So as long as depth/stencil have the same load/store it should work?

Reply:
Oh, we should fix this builder so that it doesn't force us to wipe away depth-specific options when the stencil is set. We don't need to do it right at this moment, but it's an easy fix and will probably bite us later if we don't fix it.